kirun-java/src/main/java/com/fincity/nocode/kirun/engine/function/system/array/ArrayToArrayOfObjects.java

```diff
@@ -0,0 +1,72 @@
+package com.fincity.nocode.kirun.engine.function.system.array;
+
+import java.util.List;
+import java.util.Map;
+
+import com.fincity.nocode.kirun.engine.json.schema.Schema;
+import com.fincity.nocode.kirun.engine.model.EventResult;
+import com.fincity.nocode.kirun.engine.model.FunctionOutput;
+import com.fincity.nocode.kirun.engine.model.Parameter;
+import com.fincity.nocode.kirun.engine.runtime.FunctionExecutionParameters;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+public class ArrayToArrayOfObjects extends AbstractArrayFunction {
+
+    private static final String VALUE = "value";
+
+    public static final String KEY_NAME = "keyName";
+
+    public static final String ARRAY_TO_OBJECTS = "ArrayToObjects";
+
+    protected ArrayToArrayOfObjects() {
+        super(
+                ARRAY_TO_OBJECTS,
+
+                List.of(PARAMETER_ARRAY_SOURCE, Parameter.of(KEY_NAME, Schema.ofString(KEY_NAME), true)),
+
+                EVENT_RESULT_ARRAY);
+    }
+
+    @Override
+    protected FunctionOutput internalExecute(FunctionExecutionParameters context) {
+
+        var source = context.getArguments().get(PARAMETER_ARRAY_SOURCE.getParameterName()).getAsJsonArray();
+
+        var keys = context.getArguments().get(KEY_NAME).getAsJsonArray();
+
+        JsonArray arr = new JsonArray();
+
+        if (source.size() == 0)
+
+            return new FunctionOutput(
+                    List.of(EventResult.outputOf(Map.of(EVENT_RESULT_ARRAY.getName(), new JsonArray()))));
```

Comment:
This i have updated in the js please check, then change the test case as well

Reply:
updated from output to result